components/service/pocket/src/main/java/mozilla/components/service/pocket/ext/ConceptFetch.kt

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.ext
+
+import android.support.annotation.WorkerThread
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import mozilla.components.concept.fetch.isSuccess
+import mozilla.components.service.pocket.logger
+import java.io.IOException
+
+// extension functions for :concept-fetch module.
+
+/**
+ * @return returns the string contained within the response body for the given [request] or null, on error.
+ */
+@WorkerThread // synchronous network call.
+internal fun Client.fetchBodyOrNull(callingMethod: String, request: Request): String? {
+    val response: Response? = try {
+        fetch(request)
+    } catch (e: IOException) {
+        logger.debug("$callingMethod: network error", e)
```

Comment:
I wonder why you pass in `callingMethod` - won't that be part of the stacktrace already?